src/installer/installer.go

```diff
@@ -465,23 +469,29 @@ func (i *installer) waitForController() error {
 		controllerPod := common.GetPodInStatus(kc, assistedControllerPrefix, assistedControllerNamespace,
 			map[string]string{"job-name": assistedControllerPrefix}, v1.PodRunning, i.log)
 		if controllerPod != nil {
+
 			// uploading logs here cause we will finish installing bootstrap in couple of seconds
 			// just didn't wanted to start handling this logic in other place
 			// controller must be running for at least couple of minutes before this code so it will cover 90% of the cases
 			// if we will see that it is not enough, we will change the place.
 			err = common.UploadPodLogs(kc, i.inventoryClient, i.ClusterID, controllerPod.Name, assistedControllerNamespace, common.ControllerLogsSecondsAgo, i.log)
```

Comment:
I think this method needs some refactoring as well.
The log says that we fail to upload while we suspect that we failed to get the log from kube-apiserver.

Reply:
this method returns error, in the case we saw:
"Failed to get logs of pod assisted-installer-controller-p7pjv: Get \"https://192.168.50.122:10250/containerLogs/assisted-installer/assisted-installer-controller-p7pjv/assisted-installer-controller?sinceSeconds=3600\": open /etc/kubernetes/secrets/kube-apiserver-to-kubelet-client.crt: no such file or directory"

Reply:
than we just wrap it with one more message that is relevant for our specific case. We can though just print the error that returns from UploadPodLogs